daemon/algod/api/algod.oas2.json

```diff
@@ -2975,6 +2975,44 @@
         }
       }
     },
+    "AssetHoldingReference": {
+      "description": "References an asset held by an account.",
+      "type": "object",
+      "required": [
+        "account",
+        "asset"
+      ],
+      "properties": {
+        "account": {
+          "description": "Address of the account holding the asset.",
+          "type": "string",
+          "x-algorand-format": "Address"
+        },
+        "asset": {
+          "description": "Asset ID of the holding.",
+          "type": "integer"
```

Comment:
Some integers in oas2 are annotated `"x-algorand-format": "uint64"`. When do we do that?

Reply:
TODO

Reply:
As of now, the only difference I'm aware of when adding `"x-algorand-format": "uint64"` is that the Java SDK will use a `BigInteger` instead of long.

See https://github.com/algorand/generator/blob/de85c86f7d3ffbfcf473865b70ad3573d2eed058/src/main/java/com/algorand/sdkutils/generators/OpenApiParser.java#L181-L183

We appear to not use this for app/asset IDs, probably because Java's long is good enough to handle those values.

Reply:
I suppose Java's `long` is good enough for naming existing assets and apps for a very long time. However, if we are returning an arbitrary value that was "touched" it could be that someone looked up a high numbered asset (that doesn't exist - but it's still legal to look up). So for that we need uint64

Reply:
Good point, these referenced values are not necessarily real app/asset IDs. I also confirmed that attempting to decode a value outside of the Java `long` range results in an error.

So I updated all possible fields which may contain an arbitrary uint64 with this annotation in 032e7d802.

(Note I left `BoxReference.app` alone, as only a real app attempting to access a box can populate this field.)

Reply:
cc @ahangsu in case you also had questions about this annotation